### PR TITLE
[W4.4c][W15-A1] Jackie Tan Jin Wei

### PR DESCRIPTION
**Pull request by freesoup**
Includes 4.3D

### CODE REVIEW

File: doc/UserGuide.md [hunk 1]
```diff
@@ -44,6 +44,9 @@ Format: `add NAME [p]p/PHONE_NUMBER [p]e/EMAIL [p]a/ADDRESS [t/TAG]...`
 > be seen using the `viewall` command.
 > 
 > Persons can have any number of tags (including 0)
+> Addresses are further split into multiple subclasses
```
Comment (okkhoy) [inline doc/UserGuide.md]: :+1:

File: doc/UserGuide.md [hunk 2]
```diff
@@ -44,6 +44,9 @@ Format: `add NAME [p]p/PHONE_NUMBER [p]e/EMAIL [p]a/ADDRESS [t/TAG]...`
...
+> mainly `Block` `Street` `Unit` and `Postal`. The input of
+> each subclass must be seperated by a ", ".
```
Comment (okkhoy) [inline doc/UserGuide.md]: you could update the example to show the correct address format

File: src/seedu/addressbook/Main.java [hunk 3]
```diff
@@ -84,9 +84,13 @@ private void runCommandLoopUntilExitCommand() {
...
             String userCommandText = ui.getUserCommand();
             command = new Parser().parseCommand(userCommandText);
-            CommandResult result = executeCommand(command);
-            recordResult(result);
-            ui.showResultToUser(result);
+            try {
+                CommandResult result = executeCommand(command);
+                recordResult(result);
+                ui.showResultToUser(result);
+            } catch (Exception e) {
```
Comment (okkhoy) [inline src/seedu/addressbook/Main.java]: While this works, I would suggest you to catch more specific exceptions; that way your mileage for handling exceptions is extended.

File: src/seedu/addressbook/data/person/Address.java [hunk 4]
```diff
@@ -13,6 +13,10 @@
     public static final String ADDRESS_VALIDATION_REGEX = ".+";
 
     public final String value;
+    public final Block block;
```
Comment (okkhoy) [inline src/seedu/addressbook/data/person/Address.java]: - It is not necessary to retain the `value` member. Think what can go wrong maintaining duplicate data?
- Don't make the members `public` by default. Always start with the most restrictive access modifier. If that is not sufficient, then upgrade step by step.

File: src/seedu/addressbook/data/person/Address.java [hunk 5]
```diff
@@ -27,6 +31,10 @@ public Address(String address, boolean isPrivate) throws IllegalValueException {
             throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
         }
         this.value = trimmedAddress;
+        block = new Block(trimmedAddress.split(", ")[0]);
```
Comment (okkhoy) [inline src/seedu/addressbook/data/person/Address.java]: What does our coding standard say about using numbers in array indices? [Hint! read magic numbers]

File: src/seedu/addressbook/data/person/Block.java [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+
+/**
+ * Represents an Address' Block in the address book.
+ * Guarantees: is an string.
+ */
+
+public class Block {
+    private String blockNo;
+
+    public Block(String blockNumber) {
```
Comment (okkhoy) [inline src/seedu/addressbook/data/person/Block.java]: You can include validation logic here.

File: test/expected.txt [hunk 7]
```diff
@@ -85,57 +85,57 @@
 || Enter command: || [Command entered:  add Valid Name p/12345 e/valid@e.mail a/valid, address t/invalid_-[.tag]
 || Tags names should be alphanumeric
 || ===================================================
-|| Enter command: || [Command entered:  add Adam Brown p/111111 e/adam@gmail.com a/111, alpha street]
-|| New person added: Adam Brown Phone: 111111 Email: adam@gmail.com Address: 111, alpha street Tags: 
+|| Enter command: || [Command entered:  add Adam Brown p/111111 e/adam@gmail.com a/111, alpha street, #31-50, 123446]
+|| New person added: Adam Brown Phone: 111111 Email: adam@gmail.com Address: 111, alpha street, #31-50, 123446 Tags:
```
Comment (okkhoy) [inline test/expected.txt]: good to see you have updated the tests

File: src/seedu/addressbook/Main.java (hunk 3 above)
Comment (okkhoy) [inline src/seedu/addressbook/Main.java]: where are you throwing the exception though?

**Comment by okkhoy** [comment]
@freesoup 
Please create clean PRs next time!
Some comments added. Please ack & close the PR after reading comments.

**Comment by freesoup** [comment]
Thanks for the comments, will take note of the mistakes and improve the next round!

**Pull request closed without merging**